### ISSUE TITLE
Make Groups Useful: Repository-level access controls

### ISSUE THREAD

**Issue by danhorst** [projecthydra-labs/hyku#611]
Issue #421 defined the requirements for managing groups throughout the Administrative UI. It did not address how groups would integrate with system-level access controls.

Group membership facilitates two kinds of actions:
1. Interacting with the repository system itself (this issue)
2. Interacting with content stored in the repository system (another issue)

There are roles in the repository system that are granted special permissions e.g. Repository Administrator. These roles have rules hard-coded into the ability definitions the determine how access controls are enforced. There needs to be a way to map these role keys to groups. This could be stored in a YAML file or it could bubble up into a select box in the group management interface.

_Is a GUI for associating application roles and groups necessary at this point?_

**Comment by mjgiarlo** [comment]
Blocked by #608.

**Comment by danhorst** [comment]
- [ ] Make `User#groups` return an array of group keys (strings, not symbols)
	- [ ] Add `Hyku::Group#key`; unique, indexed
	- [ ] Add `Hyku::Group#keys_for` to provide key lookup
- [ ] Change how the admin role is conveyed to use the group system
	- [ ] Generate the default admin group during application setup
	- [ ] Guard the default admin group against deletion
	- [ ] ~~Override `Ability#admin?` to use a default admin group~~ This _shouldn’t_ be necessary if `User#groups` is implemented correctly. `user_groups` includes `User#groups` in [Blacklight access controls](https://github.com/projectblacklight/blacklight-access_controls/blob/master/lib/blacklight/access_controls/ability.rb#L97).
- [ ] Ensure `current_ability.user_groups` returns the appropriate payload
